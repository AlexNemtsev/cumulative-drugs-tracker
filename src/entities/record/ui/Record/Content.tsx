import { Box, ContextMenu, Flex, Separator, Text } from '@radix-ui/themes';

import type { RecordType } from '@/shared/types/Record';

import styles from './Content.module.css';

type Props = {
  record: RecordType;
};

export const Content = (props: Props) => {
  const {
    record: { datetime, dose },
  } = props;

  const date = new Date(datetime);

  return (
    <ContextMenu.Trigger>
      <Box className={styles.container}>
        <Flex direction="column" gap="2">
          <Text size="6" weight="medium">
            {date.toLocaleDateString('ru-RU')}
          </Text>
          <Flex justify="between">
            <Text size="6" weight="medium">
              {date.toLocaleTimeString('ru-RU', {
                hour: '2-digit',
                minute: '2-digit',
                hour12: false,
              })}
            </Text>
            <Text size="6" weight="medium">
              {dose} мг
            </Text>
          </Flex>
        </Flex>
        <Separator size="4" className={styles.separator} />
      </Box>
    </ContextMenu.Trigger>
  );
};
