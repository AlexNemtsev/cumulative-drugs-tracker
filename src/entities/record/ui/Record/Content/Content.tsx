import { Box, Flex, Separator, Text } from '@radix-ui/themes';

import type { RecordType } from '@/shared/types/Record';

import { container, separator } from './Content.css';

type Props = {
  record: Pick<RecordType, 'datetime' | 'dose'>;
};

export const Content = (props: Props) => {
  const {
    record: { datetime, dose },
  } = props;

  const date = new Date(datetime);

  return (
    <Box className={container} data-testid="record-content">
      <Flex direction="column" gap="2">
        <Text size="6" weight="medium">
          {date.toLocaleDateString('ru-RU')}
        </Text>
        <Flex justify="between">
          <Text size="6" weight="medium">
            {date.toLocaleTimeString('ru-RU', {
              hour: '2-digit',
              minute: '2-digit',
              hour12: false,
            })}
          </Text>
          <Text size="6" weight="medium">
            {dose} мг
          </Text>
        </Flex>
      </Flex>
      <Separator size="4" className={separator} />
    </Box>
  );
};
