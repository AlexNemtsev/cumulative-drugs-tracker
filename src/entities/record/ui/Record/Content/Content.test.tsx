import { render, screen } from '@testing-library/react';

import { Content } from './Content';

test('Content должен отрисоваться', () => {
  render(<Content record={{ dose: '16', datetime: '2025-06-11T15:48' }} />);

  const content = screen.getByTestId('record-content');

  expect(content).toMatchSnapshot();
});
