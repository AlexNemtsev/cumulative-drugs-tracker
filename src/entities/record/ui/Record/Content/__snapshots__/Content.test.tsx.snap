// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Content должен отрисоваться 1`] = `
<div
  class="rt-Box Content_container__gyj8v0"
  data-testid="record-content"
>
  <div
    class="rt-Flex rt-r-fd-column rt-r-gap-2"
  >
    <span
      class="rt-Text rt-r-size-6 rt-r-weight-medium"
    >
      11.06.2025
    </span>
    <div
      class="rt-Flex rt-r-jc-space-between"
    >
      <span
        class="rt-Text rt-r-size-6 rt-r-weight-medium"
      >
        15:48
      </span>
      <span
        class="rt-Text rt-r-size-6 rt-r-weight-medium"
      >
        16
         мг
      </span>
    </div>
  </div>
  <span
    class="rt-Separator rt-r-orientation-horizontal rt-r-size-4 Content_separator__gyj8v1"
    data-accent-color="gray"
  />
</div>
`;
